source/fundamentals/crud/read-operations/geo.txt

```diff
@@ -0,0 +1,306 @@
+   use geographic coordinate system conventions as you may need to swap your
+   longitude and latitude values. 
+
+Here are some common GeoJSON types and their corresponding positions:
+
+- ``Point``: a single position.
+- ``LineString``: a list of two or more positions, thus forming a series of line segments.
+- ``Polygon``: a list of positions in which the first and last
+  position are the same, thus enclosing some space.
+
+To learn more about the GeoJSON shapes you can use in MongoDB, see the
+:manual:`GeoJSON manual entry </reference/geojson/>`.
+
+.. external resource
+
+Index
+~~~~~
+
+To query data stored in the GeoJSON format, add the field containing
+GeoJSON data to  a ``2dsphere`` index. The following snippet creates a
+``2dsphere`` index on the ``location.geo`` field using the ``Indexes`` builder:
+
+.. code-block:: java
+
+   import static com.mongodb.client.model.Indexes.geo2dsphere;
+
+   // <MongoCollection setup code here>
+
+   collection.createIndex(geo2dsphere("location.geo"));
+
+For more information on the ``Indexes`` builder, see our 
+:doc:`guide on the Indexes builder </fundamentals/builders/indexes>`.
+
+.. guide resource
+
+Coordinates on a 2D Plane
+-------------------------
+
+You can store geospatial data using ``x`` and ``y`` coordinates on 
+a two-dimensional Euclidean plane. We refer to coordinates on a two-dimensional
+plane as "legacy coordinate pairs".
+
+Legacy coordinate pairs have the following structure:
+
+.. code-block:: json
+
+   "<field name>" : [ x, y ]
+
+Your field should contain an array of two values in which the first represents 
+the ``x`` axis value and the second represents the ``y`` axis value.
+
+Index
+~~~~~
+
+To query data stored as legacy coordinate pairs, you must add the field containing
+legacy coordinate pairs to  a ``2d`` index. The following snippet creates a
+``2d`` index on the ``coordinates`` field using the ``Indexes`` builder:
+
+.. code-block:: java
+
+   import static com.mongodb.client.model.Indexes.geo2d;
+
+   // <MongoCollection setup code here>
+
+   collection.createIndex(geo2d("coordinates"));
+
+For more information on the ``Indexes`` builder, see our 
+:doc:`guide on the Indexes builder </fundamentals/builders/indexes>`. 
+   
+For more information on legacy coordinate pairs, see the
+:manual:`MongoDB server manual page on legacy coordinate pairs </geospatial-queries/#legacy-coordinate-pairs>`.
+
+.. external resource
+
+.. note::
+
+   Spherical (``2dsphere``) and flat (``2d``) indexes support some, but
+   not all, of the same query operators. For a full list of operators
+   and their index compatibility, see the
+   :manual:`manual entry for geospatial queries </geospatial-queries/#geospatial-query-operators>`.
+
+   .. external resource
+
+Geospatial Queries
+------------------
+
+Geospatial queries consist of query operators with GeoJSON query parameters.
+
+Query Operators
+~~~~~~~~~~~~~~~
```

Comment:
It feels like this can be combined into "Geospatial Query Operators"

Reply:
Thanks for the note. After trying to rewrite this in line with your edit, I would like to defend the current structure. Defense:

- Current structure allows GeoJSON package to be discussed separately from the Filters class
- Current structure has side right TOC mirror structure of a query (query operator and parameter comprising a completed query). 

I think breaking this section into two h1 sections ("Geospatial Query Operators" and "Geospatial Query Parameters") would less explicitly communicate the relationship between these two ideas.

Let me know if I am missing the mark in regards to your feedback, and as this comment is not intuitive to me, I think I might benefit from discussing it synchronously in a short call or over slack.